### ISSUE TITLE
Mismatch in RoPE and RMSNorm Ordering Between TorchTune and HuggingFace Qwen3 Implementations

### ISSUE THREAD

**Issue by athms** [pytorch/torchtune#2809]
Hi!

I noticed that the logits produced by TorchTune's Qwen3 implementation differ from those generated by the HuggingFace Qwen3 model.

It appears the discrepancy stems from the relative ordering of RoPE and RMSNorm in the attention module:

- In torchtune, [{q,k}_norm is applied after RoPE](https://github.com/pytorch/torchtune/blob/23b3f7b421ff891c782d021021fed328c6509adc/torchtune/modules/attention.py#L242).
- In HuggingFace's transformer, [{q,k}_norm is applied before RoPE](https://github.com/huggingface/transformers/blob/81799d8b556b3c810ed314187674bc439c0582b4/src/transformers/models/qwen3/modeling_qwen3.py#L206).

This might be a broader issue, as torchtune uses a unified TransformerDecoder implementation for many Transformer-based models, which potentially differ in the exact details of their forward passes.

**Comment by pbontrager** [comment]
Thank you for tracking this down. As far as I know, it's standard to apply qk norm after positional embeddings, but it seems you're right that it's done differently for Qwen3. [vLLM](https://github.com/vllm-project/vllm/blob/b6553be1bc75f046b00046a4ad7576364d03c835/vllm/model_executor/models/qwen3.py#L144) also matches the HF implementation.

While torchtune does primarily use shared components, every model can use it's own custom component. To solve this, the `torchtune/modules/attention.py:MultiHeadAttention` module should be copied to the qwen3 folder, modified to do qk_norm before rope, and then the qwen3 component builders should import the new attention module instead of the shared one.

**Comment by krammnic** [comment]
High-priority... I will open PR
